### ISSUE TITLE
Gemini calls failing - regression with pydantic >2.9.2

### ISSUE THREAD

**Issue by barapa** [Mirascope/mirascope#677]
### Description

Example script:
```py
from google.generativeai.types.generation_types import GenerationConfig
from mirascope.core import gemini, prompt_template
from mirascope.core.base import BaseCallResponse


@prompt_template("""
    System: You are a helpful assistant.
    
    User: Say hello!
""")
async def hello_prompt(): ...


async def test_gemini_call():
    """Test basic Gemini functionality."""
    generation_config = GenerationConfig(temperature=0.1, max_output_tokens=100)

    gemini_call = gemini.call(
        "gemini-1.0-pro-001",
        call_params={
            "generation_config": generation_config,
        },
    )(hello_prompt)

    response: BaseCallResponse = await gemini_call()  # type: ignore
    print("Test response:", response.content)
    return response.content


if __name__ == "__main__":
    import asyncio

    asyncio.run(test_gemini_call())
 ```
 
 Error:
```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/code/app/clis/ftp/test_mirascope_gemini.py", line 2, in <module>
    from mirascope.core import gemini, prompt_template
  File "/usr/local/lib/python3.12/site-packages/mirascope/__init__.py", line 7, in <module>
    from . import core as core
  File "/usr/local/lib/python3.12/site-packages/mirascope/core/__init__.py", line 28, in <module>
    from . import gemini as gemini
  File "/usr/local/lib/python3.12/site-packages/mirascope/core/gemini/__init__.py", line 9, in <module>
    from ._call import gemini_call
  File "/usr/local/lib/python3.12/site-packages/mirascope/core/gemini/_call.py", line 4, in <module>
    from ._utils import (
  File "/usr/local/lib/python3.12/site-packages/mirascope/core/gemini/_utils/__init__.py", line 5, in <module>
    from ._get_json_output import get_json_output
  File "/usr/local/lib/python3.12/site-packages/mirascope/core/gemini/_utils/_get_json_output.py", line 7, in <module>
    from ..call_response import GeminiCallResponse
  File "/usr/local/lib/python3.12/site-packages/mirascope/core/gemini/call_response.py", line 24, in <module>
    BaseCallResponse[
  File "/usr/local/lib/python3.12/site-packages/pydantic/main.py", line 806, in __class_getitem__
    submodel = _generics.create_generic_submodel(model_name, origin, args, params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generics.py", line 137, in create_generic_submodel
    created_model = meta(
                    ^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_model_construction.py", line 227, in __new__
    complete_model_class(
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_model_construction.py", line 661, in complete_model_class
    schema = cls.__get_pydantic_core_schema__(cls, handler)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/main.py", line 697, in __get_pydantic_core_schema__
    return handler(source)
           ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_schema_generation_shared.py", line 84, in __call__
    schema = self._handler(source_type)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 606, in generate_schema
    schema = self._generate_schema_inner(obj)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 875, in _generate_schema_inner
    return self._model_schema(obj)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 687, in _model_schema
    {k: self._generate_md_field_schema(k, v, decorators) for k, v in fields.items()},
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1067, in _generate_md_field_schema
    common_field = self._common_field_schema(name, field_info, decorators)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1255, in _common_field_schema
    schema = self._apply_annotations(
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 2041, in _apply_annotations
    schema = get_inner_schema(source_type)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_schema_generation_shared.py", line 84, in __call__
    schema = self._handler(source_type)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 2022, in inner_handler
    schema = self._generate_schema_inner(obj)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 880, in _generate_schema_inner
    return self.match_type(obj)
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 982, in match_type
    return self._match_generic_type(obj, origin)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1010, in _match_generic_type
    return self._union_schema(obj)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1317, in _union_schema
    choices.append(self.generate_schema(arg))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 606, in generate_schema
    schema = self._generate_schema_inner(obj)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 880, in _generate_schema_inner
    return self.match_type(obj)
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 982, in match_type
    return self._match_generic_type(obj, origin)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1022, in _match_generic_type
    return self._typed_dict_schema(obj, origin)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1451, in _typed_dict_schema
    fields[field_name] = self._generate_td_field_schema(
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1050, in _generate_td_field_schema
    common_field = self._common_field_schema(name, field_info, decorators)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1255, in _common_field_schema
    schema = self._apply_annotations(
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 2041, in _apply_annotations
    schema = get_inner_schema(source_type)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_schema_generation_shared.py", line 84, in __call__
    schema = self._handler(source_type)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 2022, in inner_handler
    schema = self._generate_schema_inner(obj)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 880, in _generate_schema_inner
    return self.match_type(obj)
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 952, in match_type
    return self._typed_dict_schema(obj, None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1451, in _typed_dict_schema
    fields[field_name] = self._generate_td_field_schema(
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1050, in _generate_td_field_schema
    common_field = self._common_field_schema(name, field_info, decorators)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1255, in _common_field_schema
    schema = self._apply_annotations(
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 2041, in _apply_annotations
    schema = get_inner_schema(source_type)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_schema_generation_shared.py", line 84, in __call__
    schema = self._handler(source_type)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 2022, in inner_handler
    schema = self._generate_schema_inner(obj)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 880, in _generate_schema_inner
    return self.match_type(obj)
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 982, in match_type
    return self._match_generic_type(obj, origin)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1010, in _match_generic_type
    return self._union_schema(obj)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1317, in _union_schema
    choices.append(self.generate_schema(arg))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 606, in generate_schema
    schema = self._generate_schema_inner(obj)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 880, in _generate_schema_inner
    return self.match_type(obj)
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 978, in match_type
    return self._dataclass_schema(obj, None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1755, in _dataclass_schema
    args = sorted(
           ^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1756, in <genexpr>
    (self._generate_dc_field_schema(k, v, decorators) for k, v in fields.items()),
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1084, in _generate_dc_field_schema
    common_field = self._common_field_schema(name, field_info, decorators)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1255, in _common_field_schema
    schema = self._apply_annotations(
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 2041, in _apply_annotations
    schema = get_inner_schema(source_type)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_schema_generation_shared.py", line 84, in __call__
    schema = self._handler(source_type)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 2022, in inner_handler
    schema = self._generate_schema_inner(obj)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 880, in _generate_schema_inner
    return self.match_type(obj)
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 982, in match_type
    return self._match_generic_type(obj, origin)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1010, in _match_generic_type
    return self._union_schema(obj)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 1317, in _union_schema
    choices.append(self.generate_schema(arg))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 606, in generate_schema
    schema = self._generate_schema_inner(obj)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 880, in _generate_schema_inner
    return self.match_type(obj)
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 991, in match_type
    return self._unknown_type_schema(obj)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pydantic/_internal/_generate_schema.py", line 509, in _unknown_type_schema
    raise PydanticSchemaGenerationError(
pydantic.errors.PydanticSchemaGenerationError: Unable to generate pydantic-core schema for <class 'google.ai.generativelanguage_v1beta.types.content.Schema'>. Set `arbitrary_types_allowed=True` in the model_config to ignore this error or implement `__get_pydantic_core_schema__` on your type to fully support it.

If you got this error by calling handler(<some type>) within `__get_pydantic_core_schema__` then you likely need to call `handler.generate_schema(<some type>)` since we do not call `__get_pydantic_core_schema__` on `<some type>` otherwise to avoid infinite recursion.

For further information visit https://errors.pydantic.dev/2.10/u/schema-for-unknown-type
error: Recipe `exec` failed on line 34 with exit code 1
```

**Comment by barapa** [comment]
This seems related to a change in pydantic. Pinning to 2.9.2 resolved the issue for me for the time being.

**Comment by willbakst** [comment]
Let me see if I can determine a minimal repro in the newest Pydantic version without Mirascope

**Comment by koxudaxi** [comment]
I can reproduce the error without Mirascope and google sdk.
Perhaps this PR is what caused the problem. https://github.com/pydantic/pydantic/pull/10576
```python
import dataclasses

from pydantic import BaseModel, ConfigDict
from typing_extensions import TypedDict


class Original:
    pass


class Schema:
    # Actual class is `protos.Schema`
    pass

@dataclasses.dataclass
class GenerationConfig:
    response_schema: Schema

class GeminiCallParams(TypedDict, total=False):
    generation_config: GenerationConfig

class BaseCallParams(BaseModel):
    model_config = ConfigDict(arbitrary_types_allowed=True)
    call_params: GeminiCallParams

```

```bash
/Users/koudai/PycharmProjects/mirascope/.venv/bin/python /Users/koudai/Library/Application Support/JetBrains/PyCharm2024.2/scratches/scratch_111.py 
Traceback (most recent call last):
  File "/Users/koudai/Library/Application Support/JetBrains/PyCharm2024.2/scratches/scratch_111.py", line 22, in <module>
    class BaseCallParams(BaseModel):
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_model_construction.py", line 226, in __new__
    complete_model_class(
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_model_construction.py", line 658, in complete_model_class
    schema = cls.__get_pydantic_core_schema__(cls, handler)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/main.py", line 697, in __get_pydantic_core_schema__
    return handler(source)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_schema_generation_shared.py", line 84, in __call__
    schema = self._handler(source_type)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 612, in generate_schema
    schema = self._generate_schema_inner(obj)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 881, in _generate_schema_inner
    return self._model_schema(obj)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 693, in _model_schema
    {k: self._generate_md_field_schema(k, v, decorators) for k, v in fields.items()},
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 693, in <dictcomp>
    {k: self._generate_md_field_schema(k, v, decorators) for k, v in fields.items()},
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 1073, in _generate_md_field_schema
    common_field = self._common_field_schema(name, field_info, decorators)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 1261, in _common_field_schema
    schema = self._apply_annotations(
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 2051, in _apply_annotations
    schema = get_inner_schema(source_type)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_schema_generation_shared.py", line 84, in __call__
    schema = self._handler(source_type)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 2032, in inner_handler
    schema = self._generate_schema_inner(obj)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 886, in _generate_schema_inner
    return self.match_type(obj)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 958, in match_type
    return self._typed_dict_schema(obj, None)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 1459, in _typed_dict_schema
    fields[field_name] = self._generate_td_field_schema(
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 1056, in _generate_td_field_schema
    common_field = self._common_field_schema(name, field_info, decorators)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 1261, in _common_field_schema
    schema = self._apply_annotations(
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 2051, in _apply_annotations
    schema = get_inner_schema(source_type)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_schema_generation_shared.py", line 84, in __call__
    schema = self._handler(source_type)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 2032, in inner_handler
    schema = self._generate_schema_inner(obj)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 886, in _generate_schema_inner
    return self.match_type(obj)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 984, in match_type
    return self._dataclass_schema(obj, None)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 1765, in _dataclass_schema
    args = sorted(
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 1766, in <genexpr>
    (self._generate_dc_field_schema(k, v, decorators) for k, v in fields.items()),
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 1090, in _generate_dc_field_schema
    common_field = self._common_field_schema(name, field_info, decorators)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 1261, in _common_field_schema
    schema = self._apply_annotations(
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 2051, in _apply_annotations
    schema = get_inner_schema(source_type)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_schema_generation_shared.py", line 84, in __call__
    schema = self._handler(source_type)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 2032, in inner_handler
    schema = self._generate_schema_inner(obj)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 886, in _generate_schema_inner
    return self.match_type(obj)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 997, in match_type
    return self._unknown_type_schema(obj)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py", line 515, in _unknown_type_schema
    raise PydanticSchemaGenerationError(
pydantic.errors.PydanticSchemaGenerationError: Unable to generate pydantic-core schema for <class '__main__.Schema'>. Set `arbitrary_types_allowed=True` in the model_config to ignore this error or implement `__get_pydantic_core_schema__` on your type to fully support it.

If you got this error by calling handler(<some type>) within `__get_pydantic_core_schema__` then you likely need to call `handler.generate_schema(<some type>)` since we do not call `__get_pydantic_core_schema__` on `<some type>` otherwise to avoid infinite recursion.

For further information visit https://errors.pydantic.dev/2.10/u/schema-for-unknown-type

Process finished with exit code 1

```

**Comment by willbakst** [comment]
This should be fixed in `v1.10`!
